internal/token/token.go

```diff
@@ -0,0 +1,96 @@
+package token
+
+import (
+	"bufio"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type TokenClient interface {
+	GetToken() string
+	CheckToken(token string) bool
+}
+
+type fileTokenClient struct {
+	path   string
+	tokens []string
+}
+
+var _ TokenClient = &fileTokenClient{}
+
+func NewFileTokenClient(path string) (*fileTokenClient, error) {
+	tokenFile, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("reading token file: %w", err)
+	}
+
+	scanner := bufio.NewScanner(tokenFile)
```

Comment:
Since these files are expected to be small, I might have just done `ioutil.ReadFile` then `Split`, but scanner is good too.

Reply:
My thinking was that if I want to iterate through each one and do `strings.TrimSpace` it doesn't really win me much.